src/stores/ui.ts

```diff
@@ -1,5 +1,19 @@
+import { computed, action, observable } from 'mobx';
 import { NavigateFn } from '@reach/router';
 
 export class UiStore {
+  @observable public isLoading = false;
+
+  @observable public errors: String[] = [];
+
+  @computed public get mostRecentErrorMessage() {
+    return this.errors[0];
+  }
+
+  @action
+  public setErrors(errors: String[]) {
+    this.errors = errors;
```

Comment:
One of the nuances to MobX is that when replacing an array, you should use the MobX function `replace`. The reason is that when initializing `@observable public errors = [];`, we are making the empty array an observable. However, when `setErrors` does this: `this.errors = errors`, now we are pointing to a completely different array and our previously observable array has been overwritten by a non-observable array. So, instead, you can do this:

```js
@action
public setErrors(errors: String[]) {
  // A MobX observable array has a "replace" method  
  this.errors.replace(errors);
}
```

Reply:
Okay, makes sense!